scio-core/src/main/scala/com/spotify/scio/coders/JIterableWrapperSerializer.scala

```diff
@@ -26,46 +26,22 @@ import com.twitter.chill.KSerializer
 
 import scala.collection.JavaConverters._
 
-/**
- * Based on [[org.apache.beam.sdk.coders.IterableLikeCoder]] and
- * [[org.apache.beam.sdk.util.BufferedElementCountingOutputStream]].
- */
 private class JIterableWrapperSerializer[T] extends KSerializer[Iterable[T]] {
 
-  private val bufferSize = 64 * 1024
-
   override def write(kser: Kryo, out: Output, obj: Iterable[T]): Unit = {
-    val buffer = new Output(bufferSize)
-    var count = 0
     val i = obj.iterator
     while (i.hasNext) {
-      if (buffer.position() >= bufferSize) {
```

Comment:
This does not guarantees that there will be enough space to put bytes from `writeClassAndObject`, which would result in `KryoException` "Buffer overflow".

After thinking about other solutions - like keeping second buffer, trying to estimate size based on previous data etc, I think the original implementation tho not optimal - works and is simple.